src/attrs_strict/_type_validation.py

```diff
@@ -205,12 +205,30 @@ def _handle_callable(attribute, callable_, expected_type):
     ):
         callable_type = empty if callable_arg is None else callable_arg.annotation
         callable_default = empty if callable_arg is None else callable_arg.default
+        callable_kind = empty if callable_arg is None else callable_arg.kind
+
         if expected_arg is None and callable_default is not empty:
             # The callable accepts more arguments than expected, but still
             # matches the expected signature because these arguments are
             # optional (have a default value)
             continue
+
+        if callable_kind == Parameter.KEYWORD_ONLY:
+            # The callable has a keyword-only parameter and either
+            # 1) expected_arg was not None, which means that the Callable type
+            #    hint tries to pass an argument to it (but Callable only works
+            #    with positional arguments)
+            # or
+            # 2) expected_arg is None and callable_default is empty, which
```

Comment:
It is a bit concerning that we exclude support for such functions altogether. They are often used to improve readability in client code, we should allow them IMO.

I understand that `typing.Callable` being positional only is causing this problem, how do people feel about emitting a warning and letting it slide? Or better - provide a flag to control this "strictness"? 💭

Reply:
To be clear, there are essentially 6 kinds of arguments:
```python
def regular(a: int) -> None: pass
def regular_default(a: int = 0) -> None: pass

def positional_only(a: int, /) -> None: pass
def positional_only_default(a: int = 0, /) -> None: pass

def keyword_only(*, a: int) -> None: pass
def keyword_only_default(*, a: int = 0) -> None: pass
```
Although `typing.Callable` can only specify types for positional arguments, the only "unsupported" argument kind is `keyword_only`. If you think about it, out of these 6 functions, `keyword_only` is the only function, which you just can't call by passing only positional arguments. It is not clear to me, which `Callable` prototype are you suggesting should accept the `keyword_only` function. `Callable[[int], None]`? `Callable[[], None]`? Neither of those are correct, since `keyword_only(3)` and `keyword_only()` are both wrong.

I don't think, that I've seen non-defaulted keyword-only arguments that frequently in production code. And if someone really needs to accept such a `Callable`, they can always annotate the attribute with plain `Callable` without the generic specialization.

```python
@attr.s
class Any:
    any_function = attr.ib(type=typing.Callable, validator=[attrs_strict.type_validator()])

@attr.s
class Strict0:
    strict_function = attr.ib(type=typing.Callable[[], None], validator=[attrs_strict.type_validator()])

@attr.s
class Strict1:
    strict_function = attr.ib(type=typing.Callable[[int], None], validator=[attrs_strict.type_validator()])
```

Which gives the following results:

|                         | Any | Strict0 | Strict1 |
|------------------------:|:---:|:-------:|:-------:|
|                 regular |  🟢 |    ❌   |    🟢   |
|         regular_default |  🟢 |    🟢   |    🟢   |
|         positional_only |  🟢 |    ❌   |    🟢   |
| positional_only_default |  🟢 |    🟢   |    🟢   |
|            keyword_only |  🟢 |    ❌   |    ❌   |
|    keyword_only_default |  🟢 |    🟢   |    ❌   |

Reply:
>  And if someone really needs to accept such a Callable, they can always annotate the attribute with plain Callable without the generic specialization.

I had forgotten about that, I can live with it 👍